pkg/operation/botanist/health_check.go

```diff
@@ -522,30 +521,16 @@ func (b *Botanist) checkControlPlane(
 	seedWorkerLister kutil.WorkerLister,
 	extensionConditions []ExtensionCondition,
 ) (*gardencorev1beta1.Condition, error) {
-	cluster, err := gardenerextensions.GetCluster(ctx, b.K8sSeedClient.Client(), b.Shoot.SeedNamespace)
-	if err != nil {
-		if apierrors.IsNotFound(err) {
-			c := checker.FailedCondition(condition, "ControlPlaneNotReady", "Control plane is not yet ready because of missing cluster resource.")
-			return &c, nil
-		}
-		b.Logger.Errorf("Failed to execute control plane health checks: %v", err)
-		return nil, err
-	}
-	// Use shoot from cluster resource here because it reflects the actual or "active" spec to determine which health checks are required.
-	// With the "confineSpecUpdateRollout" feature enabled, shoot resources have a spec which is not yet active.
-	shoot := cluster.Shoot
-	if shoot == nil {
-		return nil, errors.New("failed to execute control plane health checks because shoot is missing in cluster resource")
-	}
-
-	if exitCondition, err := checker.CheckControlPlane(shoot, b.Shoot.SeedNamespace, condition, seedDeploymentLister, seedEtcdLister, seedWorkerLister); err != nil || exitCondition != nil {
+	if exitCondition, err := checker.CheckControlPlane(effectiveShoot, b.Shoot.SeedNamespace, condition, seedDeploymentLister, seedEtcdLister, seedWorkerLister); err != nil || exitCondition != nil {
 		return exitCondition, err
 	}
-	if exitCondition, err := checker.CheckMonitoringControlPlane(b.Shoot.SeedNamespace, b.Shoot.GetPurpose() == gardencorev1beta1.ShootPurposeTesting, b.Shoot.WantsAlertmanager, condition, seedDeploymentLister, seedStatefulSetLister); err != nil || exitCondition != nil {
+
+	wantsAlertmanager := !gardencorev1beta1helper.ShootIgnoresAlerts(effectiveShoot) && effectiveShoot.Spec.Monitoring != nil && effectiveShoot.Spec.Monitoring.Alerting != nil && len(effectiveShoot.Spec.Monitoring.Alerting.EmailReceivers) > 0
```

Comment:
Probably this could be a helper function as well that can be called here and [here](https://github.com/gardener/gardener/blob/master/pkg/operation/shoot/shoot.go#L163), wdyt?